src/test/java/org/eclipse/lyo/validation/ValidationExample.java

```diff
@@ -33,30 +43,74 @@
+
+		Validator validator = new ValidatorImpl();
+        //Validate the data model against the SHACL shapes model. You can have shape and data in same model.
+		ValidationResultModel validationResult = validator.validate(dataModel, shapeModel);
+        for (ResourceModel rm : validationResult.getInvalidResources()) {
+            if (!rm.getResult().isValid()) {
+                System.out.println(rm.getResult().show());
+            } else {
+                System.out.println("Resource is Valid");
+            }
+        }
+	}
+	
+    /**
+     * Creates an OSLC resource, and validates it against its class annotations defining its SHACL Shape.
+     */
+	public static void validateOslcResourceWithShaclAnnotations() throws IllegalAccessException, IllegalArgumentException, InvocationTargetException, OslcCoreApplicationException, URISyntaxException, ParseException, DatatypeConfigurationException {
 
-        // Load the main shape model
-        Model shapeModel = ModelFactory.createDefaultModel();
-        shapeModel.read(ValidationExample.class.getResourceAsStream("/aResource-Shape.ttl"),
-                "urn:dummy", FileUtils.langTurtle);
-        shapeModel.write(System.out, "TURTLE");
+		//Create an OSLC resource
+		AResource aResource = new AResource();
+		aResource.setAnIntegerProperty(new BigInteger("101"));
+		aResource.setAStringProperty("ABC");
+		
+        //Validate the OSLC resource against the SHACL Shape annotations - as defined in the class @shacl annotations.
+		Validator validator = new ValidatorImpl();
+		ValidationResultModel validationResult = validator.validate(aResource);
+        for (ResourceModel rm : validationResult.getInvalidResources()) {
+            if (!rm.getResult().isValid()) {
+                System.out.println(rm.getResult().show());
+            } else {
+                System.out.println("Resource is Valid");
+            }
+        }
+	}
 
-        // Perform the validation of the data model against the shapes model. You can have shape
-        // and data in same model.
-        ValidationResultModel vr = new ValidatorImpl().validate(dataModel, shapeModel);
-        for (ResourceModel rm : vr.getInvalidResources()) {
+    /**
+     * Creates an OSLC resource, and validates it against its class annotations defining its OSLC Shape.
+     */
+	public static void validateOslcResourceWithOslcAnnotations() throws IllegalAccessException, IllegalArgumentException, InvocationTargetException, OslcCoreApplicationException, URISyntaxException, ParseException, DatatypeConfigurationException {
 
+		//Create an OSLC resource
+		AnOslcResource anOslcResource = new AnOslcResource();
+		anOslcResource.setAnIntegerProperty(new BigInteger("101"));
+		
+        //Validate the OSLC resource against the OSLC Shape annotations - as defined in the class @oslc annotations.
+		Validator validator = new ValidatorImpl();
+		ValidationResultModel validationResult = validator.validate(anOslcResource);
```

Comment:
Wait, how do I control whether to validate based on SHACL or OSLC Shape annotations?

Reply:
the OslcResource class has the usual OSLC annoations.
AResource has SHACL annoattions (plus the minimal @oslc annotations for jenaModelHelper).
Followup question would be: Can I combine annotations? Yes, but for validation we will choose only one of the annotations (cannot recall which) if both exists.